File: src/main/java/com/example/innkeeper/entities/Reserva.java
package com.example.innkeeper.entities;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor

public class Reserva {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)

    public Long id;

    public LocalDate dataEntrada;

    public LocalDate dataSaida;

    public BigDecimal valorTotal;

    public String status;

    @ManyToOne
    @JoinColumn(name = "hospede_id")
    private Hospede hospede;

    @ManyToOne
    @JoinColumn(name = "quarto_id")
    private Quarto quarto;
}

File: front2/app/types/reserva.ts
import { Hospede } from "./hospede";
import { Quarto } from "./quarto";

export type EnumStatusReserva = "PENDENTE" | "CONFIRMADA" | "FINALIZADA" | "CANCELADA";

export class Reserva {
  constructor(
    public id: number | null,
    public hospede: Hospede | null,
    public quarto: Quarto | null,
    public dataEntrada: string,
    public dataSaida: string,
    public valorTotal: number | null,
    public status: EnumStatusReserva
  ) {}
}

export interface ReservaFormProps {
  reservaExistente?: Reserva;
}